### DISCUSSION TITLE
How to obtain the site frequency for each amino acid site under a homogeneous model

### DISCUSSION THREAD

**Discussion by ZQ1126** [iqtree/iqtree2#247] [Q&A] [answered]
Dear Developers,

In C60-PMSF, by using "iqtree2 -st AA -nt 24 -s input.fasta -m LG+C60+F+G -pre C60_sf_input --tree-freq LG_input.treefile", I can obtain a .sitefreq file that contains frequency information at each site.

Now, without specifying a mixture model, I would like to obtain a file similar to .sitefreq (containing frequency information at each site for the amino acid matrix).
When I tried to use "iqtree2 -st AA -nt 24 -s input.fasta -m LG+FO+G -pre LG_FO_sf_input --tree-freq LG_input.treefile" for calculation, I received an error: ERROR: No mixture model was specified!

Is there a way to solve this issue? Alternatively, is there another method to obtain maximum-likelihood optimized base frequencies at each site under a homogeneous model (such as LG+FO+G)?

Best regards,
Qiang

**Comment by roblanf** [comment, marked as answer]
@ZQ1126 this is a misunderstanding of how amino acid models work.

If you specify `LG+FO`, then you are assuming that every site has the same amino acid (not base) frequencies at every site, and that this is optimised via ML. This frequency vector is reported in the IQ-TREE output files.

I suggest reading up about standard amino acid models, profile mixture models, and the PMSF model to get this clear.


Hope that helps,

Rob

**Reply by ZQ1126** [reply]
Hi Rob，
Thank you very much for your help. 
In a homogeneous model (such as LG+FO+G), we can only obtain a single set of ML optimized amino acid frequencies that apply to the entire dataset.

For the C60 model (heterogeneous model), it provides 60 fixed amino acid frequencies (essentially, there are 60 possible frequencies for each site).So, the ".sitefreq" file obtained from the C60-PMSF method contains the average frequencies at each site, which are optimized using the ML method based on these 60 fixed frequencies. 
Could you explain how each site is optimized using the ML method based on these 60 fixed frequencies?

Best regards,
Qiang

**Reply by roblanf** [reply]
Hi Qiang,

The explanation is for the C10-C60 models is here:

https://academic.oup.com/mbe/article/37/12/3616/5900673

The short answer though is that we are not optimising frequencies at each site. The C60 model is a mixture of 60 frequency classes, and the likelihood of each site is calculated under all 60 classes. Optionally you can also optimise the class weights (details in the IQ-TREE docs). The .sitefreq file contains the frequencies obtained by using the weighted (class weights) frequency vectors at each site.

Rob
